packages/@jsii/go-runtime/jsii-experimental/api.go

```diff
@@ -292,8 +292,11 @@ type ErrorResponse struct {
 	Stack *string `json:stack`
 }
 
+// Custom unmarshalling implementation for response structs. Creating new types
+// is required in order to avoid infinite recursion.
```

Comment:
I think you mentioned this offline but remind me why having just `r` as the argument  is causing recursion?

Reply:
This is a questions for @MrArnoldPalmer... He's the one who's mentioned it and I literally only copied his work (although I have been able to observe the infinite recursion... before I shamelessly copied :D)

Reply:
My understanding though is the following:

`UnmarshalJSON` is what is invoked by `json.Unmashal`, which itself is called by `unmarshalKernelResponse`, which is called by `UnmashalJSON`.

If you don't create a new type, the default unmarshaller will call the custom one (since the type matches), which will call the default unmashaller, etc... ad nauseam (I mean it'll literally barf a stack trace once it has had enough).